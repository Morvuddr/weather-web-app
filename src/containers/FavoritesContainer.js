import React from 'react';
import * as PropTypes from 'prop-types';
import Search from '../components/Search';
import FavoritesList from '../components/FavoritesList';
import { connect } from 'react-redux';
import { addNewCityAsync, deleteCity } from '../redux/favorites/actions';

class FavoritesContainer extends React.Component {

    handleSubmit = (e) => {
        e.preventDefault();
        this.props.addCity({ name: e.target.cityName.value });
    };

    handleRemove = (id) => {
        this.props.removeCity(id);
    };

    render() {
        console.log('this.props.cities', this.props.cities);
        return (
            <>
                <Search onSubmit={this.handleSubmit} />
                <FavoritesList cities={this.props.cities} onRemove={this.handleRemove} isLoading={this.props.isLoading}/>
            </>
        );
    }
}

FavoritesContainer.propTypes = {
    cities: PropTypes.arrayOf(PropTypes.object),
    isLoading: PropTypes.arrayOf(PropTypes.string),
    addCity: PropTypes.func,
    removeCity: PropTypes.func,
};

FavoritesContainer.defaultProps = {
    cities: [],
    removeCity: f => f,
    addCity: f => f,
};

const mapStateToProps = ({ favorites: { cities, isLoading } }) => ({
    cities,
    isLoading,
});

const mapDispatchToProps = {
    addCity: addNewCityAsync,
    removeCity: deleteCity,
};


export default connect(mapStateToProps, mapDispatchToProps)(FavoritesContainer);
