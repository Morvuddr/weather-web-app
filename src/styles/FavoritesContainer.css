.search{
    display: flex;
    flex-direction: column;
    margin: 10px auto 0 auto;
    height: 160px;
    width: 90%;
    justify-content: space-between;
    align-items: center;
}
.search h1 {
    text-align: left;
}
.searchInput {
    height: 35px;
    width: 160px;
    margin-right: 10px;
    background: #f0f0f0;
    text-align: center;
}
::placeholder {
    color: darkgray;
    opacity: 1;
    text-align: center;
}
.searchButton {
    height: 32px;
    width: 32px;
    border-radius: 50%;
    font-size: 16px;
    line-height: 17.6px;
    background: darkgray;
    color: white;
}

@media( min-width: 700px) {
    .search {
        height: 80px;
        flex-direction: row;
    }
}
.favCityHeader {
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin: 0 auto;
}
.deleteButton {
    height: 32px;
    width: 32px;
    border-radius: 50%;
    font-size: 16px;
    line-height: 17.6px;
    background: darkgray;
    color: white;
}
.FavList {
    width: 90%;
    margin: 0 auto;
    display: flex;
    flex-wrap: wrap;
    align-content: space-between;
    justify-content: space-between;
}
.FavoriteCity {
    width: 48%;
    height: 325px;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}
.FavoriteCity .WeatherInfo {
    width: 100%;
    height: 100%;
}
