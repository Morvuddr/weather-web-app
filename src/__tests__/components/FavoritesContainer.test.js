import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import FavoritesContainer from '../../containers/FavoritesContainer';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

let city;
let cities;
beforeEach(() => {
    city = {
        isLoading: false,
        error: false,
        weather: {
            name: 'Moscow',
            img: 'abcd',
            temperature: 'temp',
            wind: 'wind',
            cloudiness: 'cloudiness',
            pressure: 'pressure',
            humidity: 'humidity',
            location: '[location]'
        }
    };
    cities = [{
        id: '1',
        name: 'Moscow',
        img: 'abcd',
        temperature: 'temp',
        wind: 'wind',
        cloudiness: 'cloudiness',
        pressure: 'pressure',
        humidity: 'humidity',
        location: '[location]'
    }];
});

describe('FavoritesContainer container', () => {

    test('matches the snapshot when favorites cities are loaded', () => {
        const store = mockStore({
            city: city,
            favorites: {
                cities: cities,
                errors: [],
                isLoading: []
            }
        });
        const tree = renderer.create(
            <Provider store={store}>
                <FavoritesContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot when favorites cities are loading', () => {

        const store = mockStore({
            city: city,
            favorites: {
                cities: cities,
                errors: [],
                isLoading: ['1']
            }
        });

        const tree = renderer.create(
            <Provider store={store}>
                <FavoritesContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('matches the snapshot on error', () => {

         const store = mockStore({
            city: city,
            favorites: {
                cities: cities,
                errors: ['1'],
                isLoading: []
            }
        });
        const tree = renderer.create(
            <Provider store={store}>
                <FavoritesContainer />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});