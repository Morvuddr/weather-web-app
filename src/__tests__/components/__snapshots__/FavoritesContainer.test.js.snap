// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesContainer container matches the snapshot on error 1`] = `
Array [
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      >
        +
      </button>
    </form>
  </div>,
  <div
    className="FavList"
  >
    <div
      className="FavoriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="abcd"
        />
        <h1>
          temp
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          ×
        </button>
      </div>
      <div
        className="Error"
      >
        <h1>
          Не удалось загрузить данные
        </h1>
      </div>
    </div>
  </div>,
]
`;

exports[`FavoritesContainer container matches the snapshot when favorites cities are loaded 1`] = `
Array [
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      >
        +
      </button>
    </form>
  </div>,
  <div
    className="FavList"
  >
    <div
      className="FavoriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="abcd"
        />
        <h1>
          temp
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          ×
        </button>
      </div>
      <div
        className="WeatherInfo"
      >
        <div
          className="Info"
        >
          <h3>
            Ветер
          </h3>
          <h4>
            wind
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Облачность
          </h3>
          <h4>
            cloudiness
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Давление
          </h3>
          <h4>
            pressure
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Влажность
          </h3>
          <h4>
            humidity
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Координаты
          </h3>
          <h4>
            [location]
          </h4>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`FavoritesContainer container matches the snapshot when favorites cities are loading 1`] = `
Array [
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      >
        +
      </button>
    </form>
  </div>,
  <div
    className="FavList"
  >
    <div
      className="FavoriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="abcd"
        />
        <h1>
          temp
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          ×
        </button>
      </div>
      <div
        className="Loader"
      >
        <h1>
          Подождите, данные загружаются
        </h1>
        <div
          className="Spinner"
        />
      </div>
    </div>
  </div>,
]
`;
